src/Connection.php

```diff
@@ -341,9 +341,6 @@ public function awaitConnectionClose(): Protocol\MethodConnectionCloseFrame
             'filter' => function (Protocol\AbstractFrame $frame): bool {
                 if ($frame instanceof Protocol\MethodConnectionCloseFrame) {
                     return true;
-                } elseif ($frame instanceof Protocol\MethodConnectionCloseFrame) {
-                    $this->connectionCloseOk();
-                    throw new ClientException($frame->replyText, $frame->replyCode);
```

Comment:
Need to check the spec, but line 345 might actually what we would want to do on 343. But again, will confirm that

Reply:
Now that you say it - even without knowing the spec, it seems that it should be the second one. I just blindly assumed it was the first branch because that was the one being executed before the change. I'll update is as soon as I can!

Will you squash the commits or keep them when merging! I'm asking to determine if I will add a commit or rewrite the existing one.

Reply:
> Now that you say it - even without knowing the spec, it seems that it should be the second one. I just blindly assumed it was the first branch because that was the one being executed before the change. I'll update is as soon as I can!

Well, make sure to confirm it first tho 😄 .

> Will you squash the commits or keep them when merging! I'm asking to determine if I will add a commit or rewrite the existing one.

Please rewrite, I will keep the commits.

Reply:
I read up at https://www.rabbitmq.com/amqp-0-9-1-reference#connection.close

> This method indicates that the sender wants to close the connection. This may be due to internal conditions (e.g. a forced shut-down) or due to an error handling a specific method, i.e. an exception. When a close is due to an exception, the sender provides the class and method id of the method which caused the exception.
> 
> After sending this method, any received methods except Close and Close-OK MUST be discarded. The response to receiving a Close after sending Close must be to send Close-Ok.

My takeaway is that `$this->connectionCloseOk()` should indeed be called.

But I don't understand why, in the current implementation, a `ClientException` is thrown without checking the Frame's`replyCode` first. The [AMQP Reply Code Constants](https://www.rabbitmq.com/amqp-0-9-1-reference#constants) seem to indicate that an error has occurred when the `replyCode` is `>=311` and `<=541`.

The `replyCode` `200` is documented with

> Indicates that the method completed successfully. This reply code is reserved for future use - the current protocol design does not use positive confirmation and reply codes are sent only in case of an error.

So, shouldn't we check for the reply code and check if it is set, an integer and in a given range before throwing the client exception?

(Apologies for the wall of text 😅)

Reply:
Applying the above change would require to remove the first `if` branch and convert the second `elseif` branch to an `if` branch. (I hope that makes sense). I just added #153 to make this a little bit easier hier.

I made it a separate PR because it introduces a lot of changed lines, I hope that's ok!

Reply:
> Applying the above change would require to remove the first `if` branch and convert the second `elseif` branch to an `if` branch. (I hope that makes sense). I just added #153 to make this a little bit easier here.
> 
> I made it a separate PR because it introduces a lot of changed lines, I hope that's ok!

That's perfect, just merged it 👍

Reply:
> I read up at https://www.rabbitmq.com/amqp-0-9-1-reference#connection.close
> 
> > This method indicates that the sender wants to close the connection. This may be due to internal conditions (e.g. a forced shut-down) or due to an error handling a specific method, i.e. an exception. When a close is due to an exception, the sender provides the class and method id of the method which caused the exception.
> > After sending this method, any received methods except Close and Close-OK MUST be discarded. The response to receiving a Close after sending Close must be to send Close-Ok.
> 
> My takeaway is that `$this->connectionCloseOk()` should indeed be called.
> 
> But I don't understand why, in the current implementation, a `ClientException` is thrown without checking the Frame's`replyCode` first. The [AMQP Reply Code Constants](https://www.rabbitmq.com/amqp-0-9-1-reference#constants) seem to indicate that an error has occurred when the `replyCode` is `>=311` and `<=541`.
> 
> The `replyCode` `200` is documented with
> 
> > Indicates that the method completed successfully. This reply code is reserved for future use - the current protocol design does not use positive confirmation and reply codes are sent only in case of an error.
> 
> So, shouldn't we check for the reply code and check if it is set, an integer and in a given range before throwing the client exception?

Yeah probably, different PR tho, and that might play into #29 as well.

> (Apologies for the wall of text 😅 - also, I thought I already posted this 2 days ago, but found out that I started a review instead of adding a single comment ¯\_(ツ)_/¯ )

No worries!

Reply:
Thank you for merging the other PR so promptly - this made making the necessary change here so much easier! Rebased and re-committed the change! 💪🏻